chimney/src/main/scala/io/scalaland/chimney/internal/macros/TransformerMacros.scala

```diff
@@ -263,11 +263,18 @@ trait TransformerMacros extends MappingMacros with TargetConstructorMacros with
       From: Type,
       To: Type
   ): Either[Seq[TransformerDerivationError], Tree] = {
-    Right {
-      mkTransformerBodyTree0(config.derivationTarget) {
-        q"new $To($srcPrefixTree)"
-      }
-    }
+    val toValueClassMember = To.valueClassMember.toRight(
+      Seq(CantFindValueClassMember(To.typeSymbol.name.toString, From.typeSymbol.name.toString))
+    )
+
+    for {
+      toValueClassMemberType <- toValueClassMember.map(_.returnType)
+      transformerBodyTree <- resolveRecursiveTransformerBody(srcPrefixTree, config)(
+        From,
+        toValueClassMemberType
+      )
+      toCreationTree = q"new $To(${transformerBodyTree.tree})"
+    } yield mkTransformerBodyTree0(transformerBodyTree.target)(toCreationTree)
```

Comment:
```suggestion
    for {
      toValueClassMethodSymbol <- toValueClassMember
      toValueClassMemberType <- toValueClassMember.map(_.returnType)
      transformerBodyTree <- resolveRecursiveTransformerBody(srcPrefixTree, config)(
        From,
        toValueClassMemberType
      )
    } yield mkTransformerBodyTree1(
      To,
      Target.fromField(toValueClassMethodSymbol, From),
      transformerBodyTree,
      config.derivationTarget
    )(innerTree => q"new $To(${innerTree})")
```
This one is going to be quite more complicated to get it right.
It's not sufficient to just wrap recursively derived tree in `new To(...)`, as recursive tree might be lifted/partial transformer that you have to map over. For such cases `mkTransformerBodyTree1` should be used as in the snipped.